src/utils.jl

```diff
@@ -1128,3 +1128,49 @@ _argument_description(s::Symbol) =
 
 _document_argument(s::Symbol) =
     _fmt_paragraph(_argument_description(s), leadingspaces = 6 + length(string(s)))
+
+#
+# Computes the distance of the plot limit to a sample of points
+# equally spaced in the set and keeps the minimum distance found.
+#
+function _dmin_series(lim,x,y,nsamples)
+    dmin = typemax(eltype(lim))
+    step = max(1,div(length(x),nsamples))
+    for i in 1:min(nsamples,length(x))
+        isample = firstindex(x) + (i-1)*step
+        d = sum(abs2, lim .- (x[isample],y[isample]))
```

Comment:
I think we can unroll by hand and avoid the allocations here ?

Reply:
I don't think the allocations are coming from there, `lim` is a tuple, known at compile time. I've tested that. I think the allocations are coming from the fact that `xlims` are returned as `Tuple{Any,Any}`, which is beyond what I can deal with in this functions. I tried adding a function barrier but that removed 2 allocations only and was not worth the complication (the effect on time was irrelevant).